src/Arcus.Scripting.DevOps/Scripts/Set-RetainBuildFlag.ps1

```diff
@@ -0,0 +1,23 @@
+param(
+   [Parameter(Mandatory = $true)][string] $OrganizationName = $(throw "The name of the organization is required"),
+   [Parameter(Mandatory = $true)][string] $ProjectId = $(throw "ProjectId is required"),
+   [Parameter(Mandatory = $true)][string] $BuildId = $(throw "BuildId is required"),
+   [Parameter(Mandatory = $true)][string] $AccessToken = $(throw "An access Token for the DevOps API is required")
+)
+
+$retentionPayload = @{        
+  keepforever='true'
+}
+
+$requestBody = $retentionPayload | ConvertTo-Json -Depth 100
+
+$requestUri = "https://dev.azure.com/$OrganizationName/$ProjectId/_apis/build/builds/" + $BuildId + "?api-version=6.0"
```

Comment:
I didn't want to parameterize the complete 'base' uri, as I thought it would make this more complex to pass this in as a parameter.
Unless you rely on this environment variable inside the script and do not pass it via a param.

Reply:
To be honest, I don't see a point right now in parametrizing the API version ? @mbraekman  ?

Reply:
What we're typically doing, in case it's applicable, is set it as a parameter but default it to a specific version (as Stijn indicates). (for example in `Enable-AzLogicApp`)
The reason for this is that based on this approach the same script can easily be used to use the latest/another specific version of the API, without requiring any changes in the script.

However, while API versions change very often in the case of ARM templates, for instance, this doesn't seem to be the case for DevOps-related APIs. So, not quite sure if in this case, it would be very useful. 
--> So not really required in this case.

Regarding the base-URI, I would rely on the built-in variables as much as possible, as this requires less input from the client using the script, increasing user-friendliness. But, even so, we should still offer users the possibility to provide their own _custom_ values, just in case.
--> use built-in variable.

Reply:
But then I'd suggest that we do not pass in the `organizationname` parameter, but put the DevOps variable which contains the Uri inside the script.

Reply:
Yes, exactly. So, we could opt for using the built-in base URL but that this is the default value for a custom provided URL?

Reply:
I don't think this would make it more useable.  You'll have to pass in an url, consumer needs to know if it requires a trailing slash or not (ok, you can handle that in your script), but imho, it makes the whole thing a bit less user-friendly

Reply:
This one is a good example how we did it for another DevOps script:
https://github.com/arcus-azure/arcus.scripting/blob/master/src/Arcus.Scripting.DevOps/Scripts/Set-AzDevOpsArmOutputs.ps1